src/app/App.tsx

```diff
@@ -98,11 +99,19 @@ export class App extends React.Component<Props> {
 
     Linking.addEventListener('url', this.handleOpenURL)
 
+    dynamicLinks().onLink(({ url }) => this.handleOpenURL(url))
+
     const url = await Linking.getInitialURL()
     if (url) {
       await this.handleOpenURL({ url })
     }
 
+    const firebaseUrl = await dynamicLinks().getInitialLink()
```

Comment:
i think this might be nicer in the firebase sagas instead of here? what do you think? (or at least, keep the dynamic links logic together, without the Linking logic in between)

Reply:
okay, that makes sense too. i was thinking about this as being similar to firebase cloud messaging and we have some code to initialise that in the firebase saga / a firebase file where we handle opening the app based on a deep link. indeed dynamic links aren't exactly the same though and it's debatable whether grouping these is the best approach. so i am also happy to leave it here. would like to get @jeanregisser 's thoughts maybe?